Examples/Scripts/Python/seeding.py

```diff
@@ -9,97 +9,54 @@
-                vertex=acts.examples.GaussianVertexGenerator(
-                    stddev=acts.Vector4(0, 0, 0, 0), mean=acts.Vector4(0, 0, 0, 0)
-                ),
-                particles=acts.examples.ParametricParticleGenerator(
-                    p=(1 * u.GeV, 10 * u.GeV),
-                    eta=(-2, 2),
-                    phi=(0, 360 * u.degree),
-                    randomizeCharge=True,
-                    numParticles=4,
-                ),
-            )
-        ],
-        outputParticles="particles_input",
-        randomNumbers=rnd,
-    )
-
-    # Read input from input collection (e.g. Pythia8 output)
-
-    # evGen = acts.examples.RootParticleReader(
-    #     level=acts.logging.INFO,
-    #     particleCollection="particles_input",
-    #     inputDir="output",
-    #     inputFile="pythia8_particles.root",
-    # )
-
-    # Simulation
-    simAlg = acts.examples.FatrasSimulation(
-        level=acts.logging.INFO,
-        inputParticles=evGen.config.outputParticles,
-        # inputParticles=evGen.config.particleCollection,
-        outputParticlesInitial="particles_initial",
-        outputParticlesFinal="particles_final",
-        outputSimHits="simhits",
-        randomNumbers=rnd,
-        trackingGeometry=trackingGeometry,
-        magneticField=field,
-        generateHitsOnSensitive=True,
-    )
-
-    # Digitization
-    digiCfg = acts.examples.DigitizationConfig(
-        acts.examples.readDigiConfigFromJson(
-            str(
-                srcdir
-                / "Examples/Algorithms/Digitization/share/default-smearing-config-generic.json"
-            )
-        ),
-        trackingGeometry=trackingGeometry,
-        randomNumbers=rnd,
-        inputSimHits=simAlg.config.outputSimHits,
-    )
-    digiAlg = acts.examples.DigitizationAlgorithm(digiCfg, acts.logging.INFO)
+def addSeeding(
+    s: acts.examples.Sequencer,
+    trackingGeometry: acts.TrackingGeometry,
+    field: acts.MagneticFieldProvider,
+    geoSelectionConfigFile: Union[Path, str],
+    outputDir: Optional[Union[Path, str]] = None,
+    logLevel: Optional[acts.logging.Level] = None,
+) -> acts.examples.Sequencer:
+    """This function steers the seeding
+
+    Parameters
+    ----------
+    s: Sequencer
+        the sequencer module to which we add the Seeding steps (returned from addSeeding)
+    trackingGeometry : tracking geometry
+    field : magnetic field
+    geoSelectionConfigFile : Path|str, path, None
+        Json file for space point geometry selection
+    outputDir : Path|str, path, None
+        the output folder for the Root output, None triggers no output
+    logLevel : acts.logging.Level, None
+        logging level to override setting given in `s`
+    """
+
+    def customLogLevel(custom: acts.logging.Level = acts.logging.INFO):
+        """override logging level"""
+        if logLevel is None:
+            return s.config.logLevel
+        return acts.logging.Level(max(custom.value, logLevel.value))
```

Comment:
Not entirely sure I get the logic here. This is either

- The Sequencer log level, and it that case all components get this exact level
- An explicit loglevel, and in that case, the components use the more verbose out of that and what's given to this function.

Did I get this right, and if so, does that make sense?

Reply:
I agree, this is pretty arbitrary. What I was trying to do was allow to reproduce the various logging levels that were present in the old `runSeeding`. That's perhaps useful, but don't want to mess up the simple case where you just keep the same logging level specified in the `Sequencer`.

> An explicit loglevel, and in that case, the components use the more verbose out of that and what's given to this function.

Less verbose, I think. Lower numbers are more verbose, right?

Reply:
You're right yes.